server/src/main/scala/org/http4s/server/package.scala

```diff
@@ -46,12 +47,35 @@ package object server {
       service => service.compose(AuthedRequest(authUser.run))
     }
 
-    /** TODO fs2 port -- replace |||
-    def apply[Err, T](authUser: Service[Request, Err \/ T], onFailure: Kleisli[Task, AuthedRequest[Err], MaybeResponse]): AuthMiddleware[T] = { service =>
-      (onFailure ||| service)
-        .local({authed: AuthedRequest[Either[Err, T]] => authed.authInfo.bimap(err => AuthedRequest(err, authed.req), suc => AuthedRequest(suc, authed.req))})
-        .compose(AuthedRequest(authUser.run))
+    implicit def serviceChoice: Choice[Service] = new Choice[Service] {
+      override def choice[A, B, C](
+        f: Service[A, C],
+        g: Service[B, C]): Service[Either[A, B], C] = Kleisli(
+                                                               (a: Either[A, B]) =>
```

Comment:
I know this is how some formatters work, but do we want variable declarations occuring in column 64? Could Kleisli be moved to the next line after the equals sign to increase readability?